utils/zerovec/src/zerovec/mod.rs

```diff
@@ -286,14 +287,14 @@ where
     /// If you have a slice of `&[T]`s, prefer using
     /// [`Self::alloc_from_slice()`].
     #[inline]
-    pub fn new_owned(mut vec: Vec<T::ULE>) -> Self {
+    pub fn new_owned(vec: Vec<T::ULE>) -> Self {
         // Deconstruct the vector into parts
         // This is the only part of the code that goes from Vec
         // to ZeroVec, all other such operations should use this function
-        let slice: &mut [T::ULE] = &mut vec;
-        let slice = slice as *mut [_];
         let capacity = vec.capacity();
-        mem::forget(vec);
+        let len = vec.len();
+        let ptr = mem::ManuallyDrop::new(vec).as_mut_ptr();
```

Comment:
ah, this is interesting, I was trying to figure out how best to fix the problem here